### PR TITLE
Build simd images as part of the release process

### PR DESCRIPTION
**Pull request by chatton**
## Description

Task which needs to be run manually to build and push the simd docker image.

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by marbar3778 — APPROVED** [review]

File: .github/workflows/build-simd.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Create and publish simd Docker image
+
+on: workflow_dispatch
```
Comment (marbar3778) [inline .github/workflows/build-simd.yml:3]: should we make this push to master/main also can make it on pull_request to test in the PR

Comment (chatton) [reply .github/workflows/build-simd.yml:3]: I can add on pull_request and then remove before merging, ideally we'll actually just want this is the release task.

**Review by damiannolan — APPROVED** [review]
Thanks @chatton, much appreciated!

**Comment by chatton** [comment]
@damiannolan since the build/push works as expected (the image appears [here](https://github.com/cosmos/ibc-go/pkgs/container/ibc-go-simd)) I've moved the action into the release.yml file. Now any time we release an image of that tag will be built and pushed.

**Pull request merged**